src/main/java/Duke.java

```diff
@@ -1,10 +1,245 @@
+                        }
+                    }
+                    break;
+
+                }
+                case "mark": {
+                    try {
+                        int itemID = Integer.parseInt(keyboard.nextLine().trim());
+                        if ((itemID) > taskList.size()) {
+                            System.out.println("I cannot find task " + (itemID) + " as it exceeds the total tasks number");
+                        } else {
+                            System.out.println("Nice! Great job for completing this task:");
+                            taskList.get(itemID - 1).setDone();
+                            System.out.println((taskList.get(itemID - 1).toString()));
+                        }
+                    } catch (NumberFormatException e) {
+                        System.out.println("Please enter the task number");
+                    }
+                    break;
+
+                }
+                case "unmark": {
+                    try {
+                        int itemID = Integer.parseInt(keyboard.nextLine().trim()) - 1;
+                        if ((itemID + 1) > taskList.size()) {
+                            System.out.println("I cannot find task " + (itemID + 1) + " as it exceeds the total tasks number");
+                        } else {
+                            System.out.println("This item is marked as not done yet");
+                            taskList.get(itemID).setNotDone();
+                            System.out.println((taskList.get(itemID).toString()));
+                        }
+                    } catch (NumberFormatException e) {
+                        System.out.println("Please enter the task number");
+                    }
+                    break;
+                }
+                case "delete": {
+                    try {
+                        int itemID = Integer.parseInt(keyboard.nextLine().trim()) - 1;
+                        if ((itemID + 1) > taskList.size()) {
+                            System.out.println("I cannot find task " + (itemID + 1) + " as it exceeds the total tasks number");
+                        } else {
+                            System.out.println("This task is deleted from the list:");
+                            System.out.println("  " + (taskList.get(itemID).toString()));
+                            taskList.remove(itemID);
+                            System.out.println("Now you have " + taskList.size() + " tasks in the list");
+                        }
+                    } catch (NumberFormatException e) {
+                        System.out.println("Please enter the task number");
+                    }
+                    break;
+                }
+                default: {
+                    throw new DukeException("May I know what type of task this is?");
+                }
+            }
+            System.out.println("--------------------------------");
+        }
+    }
+}
```

Comment:
Don't forget to add a blank line at the end of each file! I didn't remember to do it and checkstyle flagged all my files :P